File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            return False
    return True


def alter(n, k):
    tmp = ""
    while n > 0:
        v, d = divmod(n, k)
        n = v
        tmp += str(d)
    return tmp[::-1]


def solution(n, k):
    answer = 0
    n = alter(n, k)
    n = n.split("0")
    for v in n:
        if v == "":
            continue
        if is_prime(int(v)):
            answer += 1
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        String s = Integer.toString(n, k);
        String[] split = s.split("0", -1);
        int answer = 0;
        for (String v : split) {
            if(v.isBlank()) continue;
            long longV = Long.parseLong(v);
            if (isPrime(longV)) {
                answer += 1;
            }
        }

        return answer;
    }

    public boolean isPrime(long n) {
        if (n == 1) return false;

        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                return false;
            }
        }

        return true;
    }
}